### PR TITLE
feat: support empty `password` when set `auth`

### PR DESCRIPTION
**Pull request by bluelovers**
https://github.com/unshiftio/url-parse/issues/212

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -459,6 +459,16 @@ function set(part, value, fn) {
...
       break;
 
+    case 'auth':
+      var instruction = [];
+      if (value) {
+        instruction = value.split(':');
+      }
+      url[part] = value;
+      url.username = instruction[0] || '';
+      url.password = instruction[1] || '';
```
Comment (lpinca) [inline index.js:469]:
```suggestion
      var splits = value.split(':');
      url.username = splits[0];
      url.password = splits.length === 2 ? splits[1] : '';
      url[part] = value;
```

**Comment by lpinca** [comment]
This needs tests, and for completeness setting `'username'` or `'password'` should also update `'auth'` but that can be done separately in a different PR/commit.

**Comment by lpinca** [comment]
Thank you. I'll add tests myself later.

**Review by CryptoGodfatherVA2 — COMMENTED** [review]
var5

**Pull request merged**